packages/@lwc/template-compiler/src/parser/attribute.ts

```diff
@@ -204,6 +204,12 @@ export function isAttribute(element: BaseElement, attrName: string): boolean {
         );
     }
 
+    // External custom elements default to setting data as attributes. These might be set as
+    // properties during runtime, depending on runtime heuristics.
```

Comment:
No, for each attribute, we would check if a corresponding property exists and if so, set that instead. Unfortunately we know nothing during compile time so I defaulted to attributes since we will only be setting attributes for SSR.